asset/asset.go

```diff
@@ -808,6 +808,42 @@ func (a *Asset) IsUnSpendable() bool {
 		a.Amount == 0
 }
 
+// IsSplitCommitWitness returns true if the witness is a split-commitment
+// witness.
+func IsSplitCommitWitness(witness Witness) bool {
+	return witness.PrevID != nil && len(witness.TxWitness) == 0 &&
+		witness.SplitCommitment != nil
+}
+
+// PrimaryPrevID returns the primary prev ID of an asset. This is the prev ID of
+// the first witness, unless the first witness is a split-commitment witness,
+// in which case it is the prev ID of the first witness of the root asset.
+func (a *Asset) PrimaryPrevID() *PrevID {
```

Comment:
I was hoping this conversation would come up.

I used "primary" because I think that it is special. The sort of questions I want to ask are: why does the first witness get special treatment? Is it different from the others? Why do we have a method just for the first witness?

If it is different from the others then should we recognise its difference based on whether its first or something else? Calling it "primary" gives us that flexibility.

Reply:
The first PrevOut is basically the "direct lineage" of a proof, the continuation of a straight line (e.g. a flat proof file with linked proofs). Any other inputs must be specified in the `AdditionalInputs` of a proof, so they require a different treatment.
"Primary" to me indicates a different role than the other entries (as in the primary instance of a DB cluster is the only one that is in write mode). But the first PrevID doesn't have a different role per se, it references something slightly different than the other ones. But let's not spend too much time on this discussion, I'm okay with either name.

Reply:
Ok, thanks for your thoughts. I'll add the "direct lineage" idea into the doc.